### PR TITLE
feat: Add diagnostic telemetry fields to ChatConfigRetrievalFailure

### PR DESCRIPTION
**Pull request by VijVerma**
Fixes ADO Bug #6373382

Added 4 critical diagnostic fields to improve ChatConfigRetrievalFailure telemetry for better debugging of client-side network errors:

- clientElapsedMs: Client-side elapsed time for the operation
- configuredTimeoutMs: Configured timeout value
- cancellationReason: Why the request was cancelled (timeout/cancelled/network_error/unknown)
- online: Browser online status (navigator.onLine)

Changes:
- Extended ChatSDKExceptionDetails interface with 4 optional fields
- Updated throwChatSDKError() to accept and spread diagnostic data
- Updated throwChatConfigRetrievalFailure() to accept and pass diagnostic data
- Added timing instrumentation in getChatConfig() using performance.now()
- Capture diagnostic data at both error call sites (InitializeChatSDK and InitializeLoadChatConfig)
- Added 5 new unit tests for diagnostic data functionality

All tests pass (9/9). Backward compatible - all parameters are optional. No regression risk.

# PR Details

### Description

_Include a description of the problem to be solved_

## Solution Proposed

_Detail what is the solution proposed, include links to design document if required or any other document required to support the solution_

### Acceptance criteria

_Define what are the conditions to consider the PR has achieved the intended goal_

## Test cases and evidence

_Include what tests cases were considered, any evidence of testing for future references, to identify any corner cases, etc_

### Sanity Tests

- [ ] You have tested all changes in Popout mode
- [ ] You have tested all changes in cross browsers i.e Edge, Chrome, Firefox, Safari and mobile devices(iOS and Android)
- [ ] Your changes are included in the [CHANGELOG](../CHANGE_LOG.md)

## A11y

- [ ] You have run the [Automated Accessibility Check](https://accessibilityinsights.io/docs/en/windows/getstarted/automatedchecks) and have no reported issues

***Please provide justification if any of the validations has been skipped.***

### CODE REVIEW

**Review by MugdhaSMundhe — APPROVED** [review]

**Comment by coryeurom** [comment]
## Code Review

### Overview
Adds 4 optional fields (`clientElapsedMs`, `configuredTimeoutMs`, `cancellationReason`, `online`) to `ChatSDKExceptionDetails`, threads them through `throwChatSDKError` / `throwChatConfigRetrievalFailure`, and instruments `getChatConfig()` with `performance.now()` timing plus a `createGetChatConfigDiagnosticData()` classifier mapping error codes / HTTP statuses to reasons.

### Strengths
- Backward compatible — all new params optional, all existing call sites untouched.
- Good cancellation taxonomy (`ECONNABORTED`→timeout, `ENOTFOUND`→dns_lookup_failed, etc.) — covers the realistic failure modes for the config call.
- Decent unit coverage on the exception-thrower plumbing.

### Issues / Suggestions

- **🔴 Concurrency hazard with instance-level timing fields** (`OmnichannelChatSDK.ts:171-172`, used at 2941 and 2998).
  `_lastGetChatConfigStartTime` / `_lastGetChatConfigElapsed` are stored on `this`. If two `getChatConfig()` calls overlap (e.g., `initialize()` + `initializeLoadChatConfig()` racing, or a retry path), the second call's `startTime` will clobber the first's, producing nonsense elapsed values in telemetry. These should be **local variables** captured in the `try`/`catch` closure of `getChatConfig`, then passed into the throw site — not instance state. The fact that the helper reads instance state set elsewhere is also a tight coupling smell.

- **🔴 `configuredTimeoutMs` is declared but never populated.** The interface field exists, `throwChatSDKError` accepts it, tests assert its absence — but `createGetChatConfigDiagnosticData()` never sets it. This is half-implemented dead code. Either wire it up (read from the actual axios/fetch timeout config) or drop it from the interface for now.

- **🟡 Tests don't exercise the actual error→reason mapping.** Every test in `Enhanced cancellation reason detection` constructs `diagnosticData` with a hardcoded `cancellationReason` and then asserts the exception thrower passes it through. None of them call `createGetChatConfigDiagnosticData()` with an `ECONNABORTED` / `ENOTFOUND` / etc. error and assert it returns the right reason. The interesting classification logic at lines 2944-2974 has **zero direct coverage**. Please add tests that pass real-shaped error objects to that function.

- **🟡 High-cardinality reason values.** `server_error_503`, `client_error_404` embed the status code in the dimension itself. Kusto/Aria will treat every status as a distinct category, which fragments aggregations. Consider keeping `cancellationReason` low-cardinality (`server_error`, `client_error`) and exposing the HTTP status as a separate field. Easier to `summarize count() by cancellationReason` later.

- **🟡 `performance.now()` runtime guard.** Not all targets guarantee `performance` (older Node/SSR paths). The current code crashes if it's undefined. A `typeof performance !== 'undefined' ? performance.now() : Date.now()` guard, or pulling from a shared utility, would be safer.

- **🟢 Missing newlines at EOF** on `ChatSDKExceptionDetails.ts` and the new test file (`\ No newline at end of file` in the diff). Trivial, but the repo seems to use POSIX-style trailing newlines elsewhere.

- **🟢 The PR template** is left mostly empty (sanity tests / changelog / a11y unchecked, no justification). Even an "SDK-only telemetry change, no UI surface" note would help reviewers.

### Risk: **Medium**
The concurrency bug on `_lastGetChatConfigStart/Elapsed` is the real concern — under normal single-init usage you'll never notice, but the moment something retries or runs config retrieval in parallel, the elapsed numbers in your telemetry become unreliable, defeating the purpose of the diagnostic.

File: src/OmnichannelChatSDK.ts [hunk 1]
```diff
@@ -168,6 +168,9 @@ class OmnichannelChatSDK {
     private detailedDebugEnabled = false;
     private regexCompiledForDataMasking: RegExp[] = [];
     private isEndingChat = false;
+    // ADO 6373382: Diagnostic fields for getChatConfig telemetry
+    private _lastGetChatConfigStartTime?: number;
+    private _lastGetChatConfigElapsed?: number;
```
Comment (coryeurom) [inline src/OmnichannelChatSDK.ts]: **🔴 Blocker — Concurrency hazard.**

`_lastGetChatConfigStartTime` / `_lastGetChatConfigElapsed` are stored on `this`. If two `getChatConfig()` calls overlap (e.g. `initialize()` + `initializeLoadChatConfig()` racing, or a retry path), the second call's `startTime` clobbers the first's — producing nonsense `clientElapsedMs` values in telemetry and defeating the purpose of this diagnostic.

These should be **local variables** captured in the `try` / `catch` closure of `getChatConfig`, then passed into the throw site as part of the diagnostic data — not instance state. The helper reading instance state set elsewhere is also tight coupling that makes the race hard to spot.

Suggested shape:
```ts
private async getChatConfig(...) {
    const startTime = performance.now();
    let elapsedMs: number | undefined;
    try { ... }
    catch (error) {
        elapsedMs = Math.round(performance.now() - startTime);
        // pass elapsedMs into createGetChatConfigDiagnosticData(error, elapsedMs)
        ...
    }
}
```

File: src/core/ChatSDKExceptionDetails.ts [hunk 2]
```diff
@@ -4,11 +4,20 @@
...
+ *  @param cancellationReason {string} [OPTIONAL] Reason for cancellation (timeout, user, network, etc.)
+ *  @param online {boolean} [OPTIONAL] Browser's online status at the time of failure (navigator.onLine)
  */
 interface ChatSDKExceptionDetails {
     response: string;
     message?: string;
     errorObject?: string;
+    // Tier 1 telemetry fields (ADO 6373382)
+    clientElapsedMs?: number;
+    configuredTimeoutMs?: number;
```
Comment (coryeurom) [inline src/core/ChatSDKExceptionDetails.ts]: **🔴 Blocker — Dead field.**

`configuredTimeoutMs` is declared in the interface, accepted by `throwChatSDKError`, and asserted-absent in tests — but `createGetChatConfigDiagnosticData()` never populates it. Shipping this leaves a permanently-undefined column in Aria, which is worse than not having it (people will query it and get empty results, then assume the telemetry is broken).

Either:
1. Wire it up — read from the actual axios/fetch timeout config used by `getChatConfig`, or
2. Drop it from the interface (and the docstring/tests) for now and add it in a follow-up PR when there's a real source.

**Review by coryeurom — CHANGES_REQUESTED** [review]
Two blocking issues flagged inline. Both are addressable with small changes — see comments on `OmnichannelChatSDK.ts` and `ChatSDKExceptionDetails.ts`. Non-blocking items remain in the [overall review comment](https://github.com/microsoft/omnichannel-chat-sdk/pull/563#issuecomment-4490789168).

File: src/OmnichannelChatSDK.ts (hunk 1 above)
Comment (VijVerma) [reply src/OmnichannelChatSDK.ts]: **FIXED** in commit c6538c1

  **Changes:**
  - Removed instance fields `_lastGetChatConfigStartTime` and `_lastGetChatConfigElapsed`
  - Moved timing calculation to local variables within `getChatConfig()` method
  - Timing data is attached to error object (`__chatConfigElapsedMs`) and extracted in outer catch blocks
  - Prevents timing data from being clobbered by concurrent calls

  **Verification:**
  - All 55 tests passing
  - TypeScript compilation clean
  - Manual review confirms no instance-level state for timing

File: src/core/ChatSDKExceptionDetails.ts (hunk 2 above)
Comment (VijVerma) [reply src/core/ChatSDKExceptionDetails.ts]: **FIXED** in commit b6706a6

  **Changes:**
  - Removed `configuredTimeoutMs` field from `ChatSDKExceptionDetails` interface
  - Removed from all function signatures (`throwChatSDKError`, `throwChatConfigRetrievalFailure`)
  - Updated all test cases (removed 3 references)

  **Rationale:**
  SDK doesn't have access to HTTP client timeout configuration - this field would always be undefined. Removed to avoid dead telemetry column.

  **Verification:**
  - All tests updated and passing
  - No references to `configuredTimeoutMs` remain in codebase

**Review by coryeurom — APPROVED** [review]
